test/generic/Matrix-test.jl

```diff
@@ -165,6 +176,25 @@ function test_gen_mat_constructors()
    @test x in keys(Dict(x => 1))
    @test !(y in keys(Dict(x => 1)))
 
+   # Test creation from AbstractArray without setindex!
+   A = MyTestMatrix(BigInt(3), 2)
+   S = MatrixSpace(QQ, 2, 2)
+
+   @test isa(S(A), Generic.MatSpaceElem{Rational{BigInt}})
```

Comment:
This seems unrelated to the change, was this not working in the past?

Reply:
No, it was not working in the past. This is the issue we are fixing. The original code stopped circulant matrices from SpecialMatrices working, which is what kirtsar reported.

Reply:
This test is more general of course and just verifies that any matrix that satisfies the AbstractArray interface, which is beautifully documented on the Julia website, is supported.

Reply:
> This is the issue we are fixing

Ok, but it's not obvious, because this very test passes on AA master without this PR.

Reply:
Does it pass if I change it to MatrixSpace(ZZ, 2, 2)?

Reply:
I think it now fails in master but not with this PR.

Reply:
> Does it pass if I change it to MatrixSpace(ZZ, 2, 2)?

no